src/v/serde/test/serde_test.cc

```diff
@@ -520,6 +520,7 @@ SEASTAR_THREAD_TEST_CASE(compat_test_half_field_1) {
     };
 
     auto b = serde::to_iobuf(half_field_1{.a = 1, .b = 2, .c = 3});
+    // NOLINTNEXTLINE(bugprone-use-after-move)
     BOOST_CHECK_THROW(serde::from_iobuf<small>(std::move(b)), std::exception);
```

Comment:
curious to know why the check is triggered here. is the macro expansion generating weird code?

Reply:
it's described in the commit message:

```
test: fix use after move with boost macros
boost_require_exception and boost_check_throw are

  do { ... } while (0)

macros and clang-tidy doesn't realize that the body will
only execute once. results in false positive warning for
use after move.
```